### DISCUSSION TITLE
During SSR build, I got an "ReferenceError: self is not defined"

### DISCUSSION THREAD

**Discussion by jy95** [facebook/docusaurus#8770] [Q&A] [answered]
Hello,

When improving my playground component, I encountered an issue.
When I do a "npm start", everything run smoothly but "npm build" throws me what you see in screenshot. 
![image](https://user-images.githubusercontent.com/9306961/225103235-7e699dbd-203f-4c44-b3e0-9f6e1904ecce.png)

But obviously, nothing I tried to escape SSR issue work ...

```tsx
import React from "react"
import Layout from "@theme/Layout"
import BrowserOnly from "@docusaurus/BrowserOnly"
import ErrorBoundary from "@docusaurus/ErrorBoundary"

// Monaco isn't SSR friendly

function PlaygroundComponent(): JSX.Element {
  // No SSR for the live preview
  // See https://github.com/facebook/docusaurus/issues/5747
  return (
    <ErrorBoundary
      fallback={({ error, tryAgain }) => (
        <div>
          <p>Playground component crashed because of error: {error.message}.</p>
          <button onClick={tryAgain}>Try Again!</button>
        </div>
      )}
    >
      <BrowserOnly fallback={<div>Loading...</div>}>
        {() => {
          const PlaygroundInnerComponent =
            require("@site/src/pages/components/PlaygroundInner").default
          return <PlaygroundInnerComponent />
        }}
      </BrowserOnly>
    </ErrorBoundary>
  )
}

export default function Playground(): JSX.Element {
  return (
    <Layout
      title={`Playground`}
      description="Playground of docusaurus-json-schema-plugin"
    >
      <PlaygroundComponent />
    </Layout>
  )
}
```

Any ideas ?

Thanks in advance

Source : https://github.com/jy95/docusaurus-json-schema-plugin/pull/35

**Comment by Josh-Cena** [comment]
Your revision seems good. Why doesn't it work?

**Reply by jy95** [reply]
@Josh-Cena Honestly, I don't understand : I tried several variations but same result ....
If you have an idea, feel free to suggest as I have the CI enabled on my repo ;)

**Comment by slorber** [comment, marked as answer]
`src/pages/components/PlaygroundInner.tsx` => this is considered as a Docusaurus page and will not be lazy-loaded.

Only `_` prefix allows to be ignored, you should rather rename the component folder to `_components` or move it outside the pages folder.

**Reply by jy95** [reply]
Thanks @slorber , as `src/**/components/*.tsx` matches the default [`src/pages/**/*.tsx`](https://docusaurus.io/docs/creating-pages#routing) I forgot  🤦‍♂️

Do you think adding references to the [debug routes](https://docusaurus.io/__docusaurus/debug/routes) in routings documentation like these 2 could help people avoid such mistaken ?
- https://docusaurus.io/docs/creating-pages#routing
- https://docusaurus.io/docs/create-doc 

I personaly find that the `debug mode` could have help me in such cases ...

**Reply by slorber** [reply]
I don't know, maybe just on the pages doc? I guess it's the case where this is most likely to happen.

Not sure users would have the reflex to use this debug page anyway 🤷‍♂️ You have to have the intuition for this problem in the first place to even start thinking about a routing problem.
